app/services/coupons/create_service.rb

```diff
@@ -31,18 +45,33 @@ def create(args)
 
     private
 
+    attr_reader :limitations, :organization_id
+
     def track_coupon_created(coupon)
       SegmentTrackJob.perform_later(
         membership_id: CurrentContext.membership,
         event: 'coupon_created',
         properties: {
           coupon_code: coupon.code,
           coupon_name: coupon.name,
-          organization_id: coupon.organization_id,
+          organization_id:,
         },
       )
     end
 
+    def plan_identifiers
+      key = api_context? ? :plan_codes : :plan_ids
+      limitations[key]&.compact&.uniq
+    end
+
+    def plans
+      return @plans if defined? @plans
+      return [] if plan_identifiers.blank?
+
+      finder  = api_context? ? :code : :id
```

Comment:
Layout/SpaceAroundOperators: Operator `=` should be surrounded by a single space. (https://rubystyle.guide#spaces-operators)